tests/e2e/run-e2e-tests.sh

```
@@ -0,0 +1,458 @@
+
+    # Run the authentication state integration tests
+    log_info "Running authentication state integration tests..."
+    log_info "Environment: POLKIT_E2E_MODE=$POLKIT_E2E_MODE, PAM_WRAPPER=$PAM_WRAPPER"
+
+    # Run as testuser to force polkit authentication (root bypasses auth!)
+    # Make test results directory and build dir accessible by testuser
+    chmod 777 "$TEST_RESULTS_DIR"
+    chmod -R a+rX "$BUILD_DIR"
+
+    # Use sudo -u to preserve environment variables (su - resets them)
+    if sudo -u testuser -E bash -c "cd $BUILD_DIR/tests && ./test-authentication-state-integration -v1" > "$TEST_RESULTS_DIR/auth-state-integration.log" 2>&1; then
```

Shellcheck (0.11.0)

[warning] sudo doesn't affect redirects. Use ..| sudo tee file

(SC2024)